src/Geometries/VolumePrimitives/HexagonalPrism.jl

```diff
@@ -0,0 +1,148 @@
+struct HexagonalPrism{T} <: AbstractVolumePrimitive{T, 3} ## Only upright hexagons at the moment
+    rInner::T
+    rOuter::T
+    h::T #total height
+    translate::Union{CartesianPoint{T}, Missing}#origin at middle of hexagonal prism
+end
+
+# You also have to implement the function to obtain the primitive from a config file (so an dic)
+# You also should provide a example config file containing this new primitive
+function HexagonalPrism{T}(dict::Union{Dict{Any, Any}, Dict{String, Any}}, inputunit_dict::Dict{String,Unitful.Units})::HexagonalPrism{T} where {T <: SSDFloat}
+    # ... parse values from dict to NewPrimitive{T}(...)
+    if haskey(dict, "translate")
+        translate::CartesianPoint{T} = CartesianPoint{T}(
+            haskey(dict["translate"],"x") ? geom_round(ustrip(uconvert(u"m", T(dict["translate"]["x"]) * inputunit_dict["length"] ))) : T(0),
+            haskey(dict["translate"],"y") ? geom_round(ustrip(uconvert(u"m", T(dict["translate"]["y"]) * inputunit_dict["length"] ))) : T(0),
+            haskey(dict["translate"],"z") ? geom_round(ustrip(uconvert(u"m", T(dict["translate"]["z"]) * inputunit_dict["length"] ))) : T(0))
+    else
+        translate = missing
+    end
+    rInner, rOuter = if haskey(dict["r"], "from")
+        geom_round(ustrip(uconvert(u"m", T(dict["r"]["from"]) * inputunit_dict["length"] ))), geom_round(ustrip(uconvert(u"m", T(dict["r"]["to"]) * inputunit_dict["length"])))
+    else
+        geom_round(0), geom_round(ustrip(uconvert(u"m", T(dict["r"]["to"]) * inputunit_dict["length"])))
+    end
+    h::T = ustrip(uconvert(u"m", dict["h"] * inputunit_dict["length"]))
+    return HexagonalPrism{T}(rInner, rOuter, h, translate)
+end
+
+# Auxiliary function for point in a triangle check
+function tri_area(p1::CartesianPoint{T}, p2::CartesianPoint{T}, p3::CartesianPoint{T})::T where {T}
+    return (p1.x - p3.x) * (p2.y - p3.y) - (p2.x - p3.x) * (p1.y - p3.y)
+end
+
+# is a point inside the hexagonal prism?
+function in(pt::CartesianPoint{T}, hp::HexagonalPrism{T})::Bool where {T}
+    pt = ismissing(hp.translate) ? pt : pt .- hp.translate
```

Comment:
When `hp.translate` is a CartesianVector you do not need the broadcast `.-` but only `-`